man/systemd.unit.xml

```diff
@@ -1374,11 +1374,23 @@
 
           <para>If the <varname>systemd.condition-needs-update=</varname> option is specified on the kernel
           command line (taking a boolean), it will override the result of this condition check, taking
-          precedence over any file modification time checks. If it is used
+          precedence over any file modification time checks. If the kernel command line option is used,
           <filename>systemd-update-done.service</filename> will not have immediate effect on any following
           <varname>ConditionNeedsUpdate=</varname> checks, until the system is rebooted where the kernel
           command line option is not specified anymore.</para>
-          </listitem>
+
+          <para>Note that to make this scheme effective, the timestamp of <filename>/usr/</filename> should
+          be explicitly updated after its contents are modified. The kernel will automatically update
+          modification timestamp on a directory only when immediate children of a directory are modified; an
+          update which only updates nested files will not automatically result in mtime of
+          <filename>/usr/</filename> being updated.</para>
+
+          <para>Also note that if the update method includes a call to execute appropriate post-update steps
+          itself, it should <emphasis>not</emphasis> touch the timestamp of <filename>/usr/</filename>. In a
+          typical distribution packaging scheme, packages will do any required update steps as part of the
+          installation, to make package contents immediately usable.
```

Comment:
installation → package update?

Reply:
My thinking was colored by the fact that e.g. in rpm an update is really an installation of the new version of the package + removal of the old version. I changed install→update above, but kept it here. This is because the case of a new package installation should be covered too.

Reply:
"installation or upgrade" now.